docs/guide/extending/custom_fieldtype_comparison.md

```diff
@@ -0,0 +1,136 @@
+  tags:
+    - { name: ezplatform.field_type.comparable.engine, supported_type: EzSystems\EzPlatformContentComparison\Comparison\Field\TextLine }
+
+EzSystems\EzPlatformContentComparison\Engine\NoComparisonValueEngine:
+  tags:
+    - { name: ezplatform.field_type.comparable.engine, supported_type: EzSystems\EzPlatformContentComparison\Comparison\Field\NoComparison }
+
+EzSystems\EzPlatformContentComparison\FieldType\NonComparable:
+  tags:
+    - { name: ezplatform.field_type.comparable, alias: eznoncomparable }
+
+EzSystems\EzPlatformContentComparison\FieldType\TextLine\Comparable:
+  tags:
+    - { name: ezplatform.field_type.comparable, alias: ezstring }
+```
+
+Lines 3-5 and 7-9 configure two [comparison engines](#comparison-engine) required for handling the content of the `ezstring` Field Type.
+When configuring the engines, ensure that you provided the `supported_type` tag.
+Lines 11-13 are for providing the configuration in case the Field Type versions are not comparable.
+Finally, in lines 15-17 you configure the Field Type to be comparable by using the `alias` tag.
+
+!!! note
+
+    There is a difference in tagging:
+
+    - Configure the engines with the `ezplatform.field_type.comparable.engine` tag
+    - Configure the Field Type with the `ezplatform.field_type.comparable` tag
+
+Complete the configuration with registering it in `config/services.yaml`.
+On the basis of the example provided above:
+
+``` yaml
+imports:
+    - { resource: comparable_fieldtypes.yaml }
+#...
+
+EzSystems\EzPlatformContentComparison\Service\ContentComparisonService: ~
+
+EzSystems\EzPlatformContentComparison\Service\ContentComparisonServiceInterface: '@EzSystems\EzPlatformContentComparison\Service\ContentComparisonService'
+
+EzSystems\EzPlatformContentComparison\Registry\FieldRegistry:
+    arguments: [!tagged { tag: 'ezplatform.field_type.comparable', index_by: 'alias' }]
+
+EzSystems\EzPlatformContentComparison\Registry\FieldRegistryInterface: '@EzSystems\EzPlatformContentComparison\Registry\FieldRegistry'
+
+EzSystems\EzPlatformContentComparison\Registry\ComparisonEngineRegistry:
+    arguments: [!tagged { tag: 'ezplatform.field_type.comparable.engine', index_by: 'supported_type' }]
+
+EzSystems\EzPlatformContentComparison\Registry\ComparisonEngineRegistryInterface: '@EzSystems\EzPlatformContentComparison\Registry\ComparisonEngineRegistry'
+
+```
+
+## Comparison engine
+
+The comparison engine handles the operations required for comparing the contents of Field Types.
+Note that each Field Type requires a separate comparison engine.
+
+When creating a custom engine, ensure that it implements the `eZ\Publish\SPI\Comparison\ComparisonEngine` interface:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformContentComparison\Engine;
+
+use EzSystems\EzPlatformContentComparison\Comparison\ComparisonData;
+use EzSystems\EzPlatformContentComparison\Result\ComparisonResult;
+
+interface ComparisonEngine
```

Comment:
I would move here yaml configuration for registerting engine and how to tie it with field type., so this
```
EzSystems\EzPlatformContentComparison\Engine\FieldType\TextLineComparisonEngine:
  tags:
    - { name: ezplatform.field_type.comparable.engine, supported_type: EzSystems\EzPlatformContentComparison\Comparison\Field\TextLine }
```